docs/design.md

```diff
@@ -1182,8 +1182,10 @@ If you want to enable and use encrypted images, see:
 
 By default, the whole public key is embedded in the bootloader code and its
 hash is added to the image manifest as a KEYHASH TLV entry. As an alternative
-the bootloader can be made independent of the keys by setting the
-`MCUBOOT_HW_KEY` option. In this case the hash of the public key must be
+the bootloader can be made independent of the keys (avoiding the incorporation
+of the public key into the code) by using one of the following options:
```

Comment:
for legibility I would mention these options there by names, and adds note on its main gain immediately after, then explain them in details:
```  console
... by using one of the following options: `MCUBOOT_HW_KEY` or `MCUBOOT_BUILTIN_KEY`.

Using any of these options MCUboot is independent from the public key(s).
The key(s) can be provisioned any time and by different parties.

Hardware KEYs support options details:
```